src/main/java/com/microservicio/covid/adapter/NewsAdapterImpl.java

```diff
@@ -21,34 +21,32 @@
 import org.springframework.web.util.UriComponentsBuilder;
 
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 @Component
-@Qualifier(value = "webhoseAdapter")
+@Qualifier(value = "web_hose_adapter")
 public class NewsAdapterImpl implements NewsAdapter {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(NewsController.class);
-
+    private static final String REQUEST_PARAMETER = "coronavirus casos positivos  language:spanish thread.country:AR thread.site_type:news thread.site:telefenoticias.com.ar thread.title:Coronavirus en Argentina";
+    private static final String SORT = "crawled";
+    private static final String HEADER_APPLICATION_JSON = "application/json";
+    private static final String HEADER_CONTENT_TYPE = "Content-Type";
     @Value("${news.default.url}")
     private String newsApiUrl;
     @Value("${news.default.token}")
     private String newsToken;
-    private static final String  REQUEST_PARAMETER = "coronavirus casos positivos  language:spanish thread.country:AR thread.site_type:news thread.site:telefenoticias.com.ar thread.title:Coronavirus en Argentina";
-    private static final String  SORT = "crawled";
-
-    private static final String HEADER_APPLICATION_JSON = "application/json";
-    private static final String HEADER_CONTENT_TYPE = "Content-Type";
-
     @Autowired
-    private NewsServiceDao newsServiceDao;
+    private NewsDao newsDao;
 
     @Override
     public NewsWrapper getNews(NewsDTO newsDTO) {
 
         NewsWrapper newsList = new NewsWrapper();
-        newsList.setPosts( newsServiceDao.findByPublished(newsDTO.getPublished()));
+        newsList.setPosts(newsDao.findByPublished(newsDTO.getPublished()));
 
-        if (!newsList.getPosts().isEmpty()){
+        if (!newsList.getPosts().isEmpty()) {
```

Comment:
Si la lista de post es nula, el método isEmpty lanzará NPE, se puede user CollectionUtils.isEmpty que valida por nulo y vacio a la vez. 
```suggestion
         List<News> news = newsDao.findByPublished(newsDTO.getPublished());
        if(!CollectionUtils.isEmpty(news)) {
            newsList.setPosts(newsDao.findByPublished(newsDTO.getPublished()));
        } else {
            LOGGER.info("Existed news for the date {} consulting.", newsDTO.getPublished());
            return newsList;
        }
```

Reply:
Done